hilti/toolchain/src/compiler/parser/parser.yy

```diff
@@ -21,7 +21,7 @@ namespace hilti { namespace detail { class Parser; } }
 %lex-param   {class Driver* driver}
 
 %define api.namespace {hilti::detail::parser}
-%define api.parser.class {Parser}
+%define parser_class_name {Parser} /* new name after switching to bison 3.3 -> api.parser.class */
```

Comment:
A newer Bison will emit almost the same information,
```
/spicy/hilti/toolchain/src/compiler/parser/parser.yy:24.1-34: warning: deprecated directive: '%define parser_class_name {Parser}', use '%define api.parser.class {Parser}' [-Wdeprecated]
   24 | %define parser_class_name {Parser} /* new name after switching to bis...
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | %define api.parser.class {Parser}
```

Let's reword the comment so we make it very clear from the deprecation warning that this is intentional, e.g.,
```suggestion
%define parser_class_name {Parser} /* Using deprecated directive for bison-3.0 compatibility. */
```

Reply:
Oh, that means we'd now always be getting a warning with new Bisons? That's not so great. I like #886  very much, but I don't like unnecessary warnings ...

Reply:
FTR, this is a non-fatal warning from Bison which does not break the build, even with `--enable-werror`. I think suppressing the warning would not be great either, since that could hide genuinely interesting warnings.

Reply:
I'd like to keep the build generally free of warnings so would prefer some way of conditionally compiling this based on Bison version. It looks like there's no direct way to do that unfortunately, so we might need to do some manual replacement through `sed` or such prior to compilation. Not great, but better than always being warned IMO.

Reply:
@0xxon, I updated the build setup to adjust the sources if needed. I also appended the container update #886 to this PR. PTAL.